### PR TITLE
dioxus-cli: 0.4.0 -> 0.4.1

### PR DESCRIPTION
**Pull request by ahirner**
* update src
* use testers.testVersion

## Description of changes

[Release notes](https://github.com/DioxusLabs/dioxus/releases/tag/v0.4.1).

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

```
nix shell .#dioxus-cli
dx
dx -V
dx create  # enter data
# further testing requires rust toolchain
```

### CODE REVIEW

File: pkgs/development/tools/rust/dioxus-cli/default.nix [hunk 1]
```diff
@@ -32,10 +34,11 @@ rustPlatform.buildRustPackage rec {
...
 
-  doInstallCheck = true;
-  installCheckPhase = ''
-    $out/bin/dx --version | grep "dioxus ${version}"
-  '';
+  passthru.tests.version = testers.testVersion {
+    package = dioxus-cli;
+    command = "${meta.mainProgram} --version";
+    inherit version;
+  };
```
Comment (ahirner) [inline pkgs/development/tools/rust/dioxus-cli/default.nix:41]: Not sure if this is correct, namely wether `passthru.tests.version` is run as intended and a full replacement of the install check.

Comment (CathalMullan) [reply pkgs/development/tools/rust/dioxus-cli/default.nix:41]: Seems to work fine.
Tested by changing 'version' to '0.5.1'.

```
$ nix build .#dioxus-cli.passthru.tests.version
error: builder for '/nix/store/lp1c11ggf98yiwai7v9pxpb9gwn326g8-dioxus-cli-0.4.1-test-version.drv' failed with exit code 1;
       last 3 log lines:
       > Version string '0.5.1' not found!
       > The output was:
       > dioxus 0.4.1
       For full logs, run 'nix log /nix/store/lp1c11ggf98yiwai7v9pxpb9gwn326g8-dioxus-cli-0.4.1-test-version.drv'.
```

**Comment by CathalMullan** [comment]
Result of `nixpkgs-review pr 255589` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dioxus-cli</li>
  </ul>
</details>

**Comment by CathalMullan** [comment]
Result of `nixpkgs-review pr 255589` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dioxus-cli</li>
  </ul>
</details>

**Review by CathalMullan — APPROVED** [review]

**Review by xanderio — APPROVED** [review]

**Pull request merged**